spec/features/merchants/merchant_items/show_spec.rb

```diff
@@ -0,0 +1,165 @@
+    @transaction_25 = @invoice_9.transactions.create!(credit_card_number: '4140149827486249', credit_card_expiration_date: '10/22/20', result: 'failed')
+    @transaction_26 = @invoice_9.transactions.create!(credit_card_number: '4140149827486249', credit_card_expiration_date: '10/22/20', result: 'failed')
+    @transaction_27 = @invoice_9.transactions.create!(credit_card_number: '4140149827486249', credit_card_expiration_date: '10/22/20', result: 'failed')
+
+    @transaction_28 = @invoice_10.transactions.create!(credit_card_number: '4923661117104166', credit_card_expiration_date: '08/22/20', result: 'success')
+    @transaction_29 = @invoice_10.transactions.create!(credit_card_number: '4923661117104166', credit_card_expiration_date: '08/22/20', result: 'success')
+    @transaction_30 = @invoice_10.transactions.create!(credit_card_number: '4923661117104166', credit_card_expiration_date: '08/22/20', result: 'success')
+    #Customer 5 - total successful transactions = 3
+    #Customer 6
+    @transaction_31 = @invoice_11.transactions.create!(credit_card_number: '4923661117104166', credit_card_expiration_date: '08/22/20', result: 'success')
+    @transaction_32 = @invoice_11.transactions.create!(credit_card_number: '4923661117104166', credit_card_expiration_date: '08/22/20', result: 'success')
+    @transaction_33 = @invoice_11.transactions.create!(credit_card_number: '4923661117104166', credit_card_expiration_date: '08/22/20', result: 'success')
+
+    @transaction_34 = @invoice_12.transactions.create!(credit_card_number: '4923661117104166', credit_card_expiration_date: '08/22/20', result: 'success')
+    @transaction_35 = @invoice_12.transactions.create!(credit_card_number: '4923661117104166', credit_card_expiration_date: '08/22/20', result: 'success')
+    @transaction_36 = @invoice_12.transactions.create!(credit_card_number: '4923661117104166', credit_card_expiration_date: '08/22/20', result: 'success')
+    #Customer 6 - total successful transactions = 6
+  end
+
+  #6. Merchant Items show Page
+  it "Has a link for every Item" do
+    visit merchant_items_path(@merchant_1)
+
+    within("div#merchant-header") do
+      expect(page).to have_content(@merchant_1.name).once
+      expect(page).to_not have_content(@merchant_2.name)
+    end
+
+    within("div#merchant-items") do
+      expect(page).to have_link("#{@item_1.name}")
+      expect(page).to have_link("#{@item_2.name}")
+      expect(page).to have_link("#{@item_3.name}")
+      expect(page).to have_link("#{@item_4.name}")
+      expect(page).to have_link("#{@item_5.name}")
+      expect(page).to have_link("#{@item_6.name}")
+      expect(page).to have_link("#{@item_7.name}")
+      expect(page).to have_link("#{@item_8.name}")
+      expect(page).to have_link("#{@item_9.name}")
+      expect(page).to have_link("#{@item_10.name}")
+    end
+  end
+
+  it "After clicking on a item name I am taken to the item show page" do
+    visit merchant_item_path(@merchant_1, @item_1.id)
```

Comment:
For this test, I think they want us to start on the merchant_items index page, then click a link to a merchant_item show page.

The validations all look good, though.

Maybe replace line 154 with 

```
visit merchant_items_path(@merchant_1)

click_link("#{@item_1.name}")

expect(current_path).to eq(merchant_item_path(@merchant_1, @item_1.id))
```

Reply:
Good point, I moved that test block and added ur suggestions